mri/operators/linear/wavelet.py

```diff
@@ -57,60 +67,89 @@ def get_coeff(self):
     def set_coeff(self, coeffs):
         self.transform.analysis_data = coeffs
 
+    def _op(self, data):
+        if isinstance(data, np.ndarray):
+            data = pysap.Image(data=data)
+        self.transform.data = data
+        self.transform.analysis()
+        coeffs, coeffs_shape = flatten(self.transform.analysis_data)
+        return coeffs, coeffs_shape
+
     def op(self, data):
         """ Define the wavelet operator.
-
         This method returns the input data convolved with the wavelet filter.
-
         Parameters
         ----------
         data: ndarray or Image
             input 2D data array.
-
         Returns
         -------
         coeffs: ndarray
             the wavelet coefficients.
         """
-        if isinstance(data, np.ndarray):
-            data = pysap.Image(data=data)
-        self.transform.data = data
-        self.transform.analysis()
-        coeffs, self.coeffs_shape = flatten(self.transform.analysis_data)
+        if self.n_coils > 1:
+            coeffs, self.coeffs_shape = zip(*Parallel(n_jobs=self.n_jobs,
+                                                      backend=self.backend,
+                                                      verbose=self.verbose)(
+                delayed(self._op)
+                (data[i])
+                for i in np.arange(self.n_coils)))
+            coeffs = np.asarray(coeffs)
+        else:
+            coeffs, self.coeffs_shape = self._op(data)
         return coeffs
 
-    def adj_op(self, coeffs, dtype="array"):
+    def _adj_op(self, coeffs, coeffs_shape, dtype="array"):
         """ Define the wavelet adjoint operator.
-
         This method returns the reconsructed image.
-
         Parameters
         ----------
         coeffs: ndarray
             the wavelet coefficients.
         dtype: str, default 'array'
             if 'array' return the data as a ndarray, otherwise return a
             pysap.Image.
-
         Returns
         -------
         data: ndarray
             the reconstructed data.
         """
-        self.transform.analysis_data = unflatten(coeffs, self.coeffs_shape)
+        self.transform.analysis_data = unflatten(coeffs, coeffs_shape)
```

Comment:
Why can't we keep it as an instance property?

Reply:
for multichannel we need accumulated coeff. Note that we do keep it as instance property, just not here but later